internal/biz/user.go

```diff
@@ -0,0 +1,97 @@
+package biz
+
+import (
+	v1 "demo/api/user/v1"
+	"demo/internal/data/models"
+	"demo/internal/data/repositories/base"
+)
+
+type UserHandler struct {
+	userRepository base.UserRepository
+}
+
+func NewUserHandler(repo base.UserRepository) *UserHandler {
+	return &UserHandler{repo}
+}
+func convertFrom(userDto *v1.UserDto) models.UserModel {
+	return models.UserModel{
+		Name:             userDto.Name,
+		Age:              userDto.Age,
+		Email:            userDto.Email,
+		CurrentAddress:   userDto.GetCurrentAddress(),
+		PermanentAddress: userDto.GetPermanentAddress(),
+	}
+}
+func convertTo(userModel models.UserModel) *v1.UserDto {
+	return &v1.UserDto{
+		Id:               int32(userModel.ID),
+		Name:             userModel.Name,
+		Age:              userModel.Age,
+		Email:            userModel.Email,
+		CurrentAddress:   userModel.CurrentAddress,
+		PermanentAddress: userModel.PermanentAddress,
+	}
+}
+func update(userDto *v1.UserDto, userModel *models.UserModel) {
+	userModel.Name = userDto.Name
+	userModel.Age = userDto.Age
+	userModel.Email = userDto.Email
+	userModel.CurrentAddress = userDto.CurrentAddress
+	userModel.PermanentAddress = userDto.PermanentAddress
+}
+func (userHandler *UserHandler) Create(userDto *v1.UserDto) (*v1.CreateUserReply, error) {
+	userModel := convertFrom(userDto)
+	user, err := userHandler.userRepository.Create(&userModel)
+	if err != nil {
+		return nil, err
+	}
+	return &v1.CreateUserReply{
+		UserResponse: &v1.UserResponse{
+			UserDto: convertTo(*user),
+		},
+	}, nil
+}
+
+func (userHandler *UserHandler) Update(userDto *v1.UserDto) (*v1.UpdateUserReply, error) {
+	userModel, err := userHandler.GetById(userDto.GetId())
```

Comment:
you can directly use the repository function here.

Reply:
why handler function is also checking for non nil err, it is also calling the repo function only internally